src/consumer.ts

```diff
@@ -205,6 +221,10 @@ export class Consumer extends TypedEventEmitter {
     response: ReceiveMessageCommandOutput
   ): Promise<void> {
     if (hasMessages(response)) {
+      response.Messages.forEach(({ MessageId }) =>
+        this.messagesInQueue.push(MessageId)
```

Comment:
I hope you are ok with me commenting on your PR :)
----

U never use the message ids from `messagesInQueue`, I guess u can convert it to a counter

Something like:
on messages received:
```
this.messagesInFlight = response.Messages.length
```

on polling:

```
if (this.messagesInFlight) {
  // skip this polling
}
```

on acknolendge

```
if (!this.messagesInFlight) {
 // handle desync
}

this.messagesInFlight--;
```

Reply:
Hey, I did think about doing this, however, I would like to allow users to get the messages that are currently in the queue and also clear any messages from the queue should they need to, for added compatibility should it be required.